app/models/concerns/package_filtering.rb

```diff
@@ -39,12 +39,13 @@ def filter(options={})
       states = (options['state'] || '').strip.split(',') || []
       location = options['location']
       associated_package_types = options['associated_package_types']
+      storage_type_name = options['storage_type_name']
 
       query = where.not(inventory_number: nil)
       state_filters = states & %w[in_stock received designated dispatched]
       query = query.where_states(state_filters) if state_filters.any?
       query = query.filter_by_location(location) if location.present?
-      query = query.filter_by_package_types(associated_package_types) if associated_package_types
+      query = query.filter_by_package_types(associated_package_types, storage_type_name) if associated_package_types
```

Comment:
@abhinav098 does the `storage_type` filter need to be inside the `filter_by_package_types` methods?

Feels like it would be better to have

`query = query.filter_by_storage_types(storage_type_name) if storage_type_name.present?`

Reply:
sure @steveyken 👍